### PR TITLE
[10.0] do not run cron jobs during install/upgrade

### PR DESCRIPTION
**Pull request by rco-odoo**
Also add some useful information in return.
This is somehow needed for Odoo.sh

Safer alternative to #20580

### CODE REVIEW

**Comment by rco-odoo** [comment]
@odony @amigrave I rewrote the last commit to make the changes simpler.

File: odoo/addons/base/ir/ir_cron.py [hunk 1]
```diff
@@ -171,41 +171,53 @@ def _process_job(cls, job_cr, job, cron_cr):
...
                                   ORDER BY priority""")
                     jobs = cr.dictfetchall()
-                else:
-                    _logger.warning('Skipping database %s as its base version is not %s.', db_name, BASE_VERSION)
         except psycopg2.ProgrammingError, e:
             if e.pgcode == '42P01':
                 # Class 42 — Syntax Error or Access Rule Violation; 42P01: undefined_table
                 # The table ir_cron does not exist; this is probably not an OpenERP database.
                 _logger.warning('Tried to poll an undefined table on database %s.', db_name)
+                error = 'undefined table'
```
Comment (odony) [inline odoo/addons/base/ir/ir_cron.py]: Returning single words seems simpler in terms of API, for example `'database'` to indicate that this database is in an unknown state.

**Comment by rco-odoo** [comment]
@odony thanks for the review. I modified the PR according to your remarks.

**Comment by rco-odoo** [comment]
ping @odony @amigrave

File: odoo/addons/base/ir/ir_cron.py [hunk 2]
```diff
@@ -170,105 +177,107 @@ def _process_job(cls, job_cr, job, cron_cr):
             cron_cr.commit()
 
     @classmethod
-    def _acquire_job(cls, db_name):
+    def _process_jobs(cls, db_name):
         """ Try to process all cron jobs.
```
Comment (mreficent) [inline odoo/addons/base/ir/ir_cron.py]: Shouldn't the descriptions of `_acquire_job` and `_process_jobs` methods be a bit different?

**Comment by blaggacao** [comment]
@rco-odoo Thank you! You read my mind! I was just implementing something very similar two weeks ago... FYI, I implemented additionally a maintenance worker with lower priority (nice 15) than the cron worker and did only let the user "schedule". This leaves far better resource control for resource intensive installations such as a huge chart of account or something similar...

When an installation breaks because of a resource limit, then parent left and right for menus are usually not calculated which leaves the UI in some inconsistent state.

With a maintenance worker, this can be separately controlled...

**Comment by blaggacao** [comment]
Here is my gist:
https://gist.github.com/blaggacao/45df324285872eac53b34ef561921a76

**Comment by blaggacao** [comment]
I did use a `pg_advisory_lock` for the locking between the workers... I think it is even more appropriate for the inter process communication, although a query just does the job. `pg_advisory_loocks` are stated to be faster.

**Pull request merged**